source/world/level/Level.cpp

```diff
@@ -69,7 +69,7 @@ Level::~Level()
 		
 		//you better HOPE this is freed by Minecraft!
 		//Really should have used shared pointers and stuff.
-		if (!pEnt->isLocalPlayer())
+		//if (!pEnt->isLocalPlayer())
```

Comment:
Warning! This hack was added in for a reason -- it's possible that m_pLocalPlayer or m_pMobPersp may have a stale reference after this which would be absolutely terrible

Reply:
How is that possible? From what I can tell, Level is deconstructed in SavingWorldScreen::tick(), then m_pMobPersp and m_pLocalPlayer are set to nullptr.

Reply:
Wasn't the case when I added that line. I'll take a look later and see when this happens.

Reply:
I've changed this to isPlayer, since the NetworkHandlers are typically in charge of freeing the Player pointers in NetworkPlayer. Originally, the server would crash if a player left who had died, since their entity would be removed by the Level but was still present in the NetworkPlayer.